Engine/DataFeeds/UniverseSelection.cs

```diff
@@ -156,14 +157,30 @@ public SecurityChanges ApplyUniverseSelection(Universe universe, DateTime dateTi
                 }
             }
 
+            var keys = _pendingSecurityAdditions.Keys;
+            if (keys.Any() && keys.Single() != dateTimeUtc)
+            {
+                // if the frontier moved forward then we've added these securities to the algorithm
+                _pendingSecurityAdditions.Clear();
+            }
+
+            Dictionary<Symbol, Security> pendingAdditions;
+            if (!_pendingSecurityAdditions.TryGetValue(dateTimeUtc, out pendingAdditions))
```

Comment:
So this whole thing is kind of hacky, *but works* and is contained within this single file which is kind of nice. A more holistic approach would be to refactor everywhere in the solution where we try to get a security to use some common abstraction which would also be able to track these 'pending' security changes

Reply:
I like the single file change too! Nice fix.

Reply:
Could there be any issues from clearing the pending array using the Utc - day change over? Perhaps this should be something like "Latest Universe TimeZone Day"?